core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -595,6 +597,9 @@ oppia.controller('StateResponses', [
             $scope.trainingDataFeedback = '';
             $scope.trainingDataOutcomeDest = '';
 
+            // Inject textInputRulesService dynamically
+            textInputRulesService = $injector.get('textInputRulesService');
```

Comment:
Hmm, the names of the rules services are pretty mechanically generated: take the interaction ID, lowercase the first char, and append "RulesService". So we could just reverse that?

The interaction ID should be retrieved via stateInteractionIdService.savedMemento.